__tests__/table_moveRow.js

```diff
@@ -0,0 +1,489 @@
+          )
+        )
+      );
+
+      const newTr = moveRow(1, 2)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(td(p('4')), td(p('5')), td(p('6'))),
+            row(
+              td({ rowspan: 2 }, p('---merged-row----')),
+              td(p('0')),
+              td(p('1'))
+            ),
+            row(td(p('2')), td(p('3')))
+          )
+        )
+      );
+    });
+
+    it('should not move rows for multi rows merged', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(
+              td({ rowspan: 2 }, p('---merged-row----')),
+              td(p('0')),
+              td(p('1'))
+            ),
+            row(td(p('2')), td({ rowspan: 2 }, p('---merged-row----'))),
+            row(td(p('4')), td(p('5')))
+          )
+        )
+      );
+
+      const newTr = moveRow(1, 2)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(
+              td({ rowspan: 2 }, p('---merged-row----')),
+              td(p('0')),
+              td(p('1'))
+            ),
+            row(td(p('2')), td({ rowspan: 2 }, p('---merged-row----'))),
+            row(td(p('4')), td(p('5')))
+          )
+        )
+      );
+    });
+
+    it('should not move rows between two merged rows', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(
+              td({ rowspan: 2 }, p('---merged-row----')),
+              td(p('0')),
+              td(p('1'))
+            ),
+            row(td(p('2')), td({ rowspan: 2 }, p('---merged-row----'))),
+            row(td(p('4')), td(p('5')))
+          )
+        )
+      );
+
+      const newTr = moveRow(0, 2)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(
+              td({ rowspan: 2 }, p('---merged-row----')),
+              td(p('0')),
+              td(p('1'))
+            ),
+            row(td(p('2')), td({ rowspan: 2 }, p('---merged-row----'))),
+            row(td(p('4')), td(p('5')))
+          )
+        )
+      );
+    });
+  });
+
+  describe('on a complex table with merged cells and rows', () => {
+    it('keep the merged content columns order', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(tdEmpty, td(p('1')), td(p('2'))),
+            row(tdEmpty, td({ colspan: 2 }, p('3'))),
+            row(tdEmpty, td(p('4')), td(p('5')))
+          )
+        )
+      );
+
+      const newTr = moveRow(1, 0)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(tdEmpty, td({ colspan: 2 }, p('3'))),
+            row(tdEmpty, td(p('1')), td(p('2'))),
+            row(tdEmpty, td(p('4')), td(p('5')))
+          )
+        )
+      );
+    });
+
+    it('should not move rows when there is not movement', () => {
```

Comment:
the test doesn't do what it says, you're actually moving rows here. I think you meant to use `rowspan=3` here